ccw.core/src/java/ccw/CCWPlugin.java

```diff
@@ -234,6 +234,17 @@ private final void logDependenciesInformation(final BundleContext context) {
 
     public CCWPlugin() { }
 
+    private volatile int bundleState = Bundle.UNINSTALLED;
+
+    /**
+     * Return if the bundle is going to stop.
+     * @return True or false.
+     */
+    public static boolean isBundleStopping() {
+        CCWPlugin plugin = CCWPlugin.getDefault();
+        return plugin != null ? plugin.bundleState == Bundle.STOPPING : false;
```

Comment:
Shouldn't we refuse to load code with withBundle() is the plugin is null? Right now your change returns false if the plugin is null, and so withBundle() will continue its work.
Also, if the bundle is stopped, or uninstall, or I don't know what, maybe we shouldn't load code as well? Not just "while" it is stopping? => What about being more explicit in the test condition: if (isBundleStopping() || plugin == null) return null; ?